app/src/main/java/com/example/demo/RequestHandler.java

```diff
@@ -0,0 +1,58 @@
+package com.example.demo;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+
+public class RequestHandler implements HttpHandler {
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+        String requestContent = getRequestContent(exchange);
+
+        requestContent = processRequest(exchange, requestContent);
+
+        sendResponse(exchange, requestContent);
+    }
+
+    private String processRequest(HttpExchange exchange, String requestContent) {
+        String method = exchange.getRequestMethod();
+        URI uri = exchange.getRequestURI();
+        String path = uri.getPath();
+
+        requestContent = getRequestContent(requestContent, method, path);
+
+        return requestContent;
+    }
+
+    private String getRequestContent(String requestContent, String method, String path) {
+        if (method.equals("POST") && path.equals("/calculations")) {
+            requestContent = "success";
+        } else if (method.equals("GET") && path.equals("/calculations")) {
+            requestContent = "success";
+        }
+
+        return requestContent;
```

Comment:
if-else를 쓰고 있다면 어디선가 불필요한 일이 벌어지고 있는 겁니다. 이 코드는 몇 가지 이슈가 있습니다.

1. 매개변수의 값을 고치고 있습니다. 이는 반드시 피해야 하는 일입니다.
2. 매개변수가 아니라 임시 변수를 사용한다면, 불필요한 임시 변수입니다.
3. 어느 조건에도 맞지 않는다면 입력 받은 값을 그대로 다시 리턴합니다. 잠정적으로 버그일 가능성이 큽니다.

단순히 이런 코드가 돼야 합니다.

```suggestion
        if (method.equals("POST") && path.equals("/calculations")) {
            return "success";
        }
        if (method.equals("GET") && path.equals("/calculations")) {
            return "success";
        }

        return "기본 값 내지는 에러";
```

Reply:
여기서 만약 url과 method를 계속해서 추가하기 보다는 다형성으로 처리 해볼 수 있지 않을까 합니다.

Reply:
다형성을 다루려면 먼저 동일한 형태의 반복으로 바꿔주는 게 좋습니다. 유사한 모양의 코드 반복 + 일치하지 않을 때의 기본 값 내지는 에러 정의가 먼저 선행돼야 하죠.